### PR TITLE
feat(test): block conversion on sync tests

### PR DESCRIPTION
**Pull request by azurwastaken**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

## What is the current behavior?

Resolves: #?

## What is the new behavior?

- Add test for block conversion

## Does this introduce a breaking change?

No

### CODE REVIEW

File: crates/client/sync/src/l2.rs [hunk 1]
```diff
@@ -247,3 +247,209 @@ pub async fn sync(
...
+    #[tokio::test]
+    async fn test_l2_block_conversion_task(test_setup: Arc<MadaraBackend>) {
+        let backend = test_setup;
+        let (updates_sender, updates_receiver) = mpsc::channel(100);
+        let (output_sender, mut output_receiver) = mpsc::channel(100);
+        let block_import =
+            Arc::new(BlockImporter::new(backend.clone(), &MetricsRegistry::dummy(), None, true).unwrap());
+        let validation = BlockValidationContext::new(backend.chain_config().chain_id.clone());
+
+        let mock_block = UnverifiedFullBlock {
```
Comment (Mohiiit) [inline crates/client/sync/src/l2.rs]: maybe we can have a helper function inside the `crates/client/block_import/src/tests/block_import_utils` which would return `UnverifiedFullBlock`. This helper might be helpful for other tests as well

File: crates/client/sync/src/l2.rs [hunk 2]
```diff
@@ -247,3 +247,209 @@ pub async fn sync(
...
+    #[tokio::test]
+    async fn test_l2_verify_and_apply_task(test_setup: Arc<MadaraBackend>) {
+        let backend = test_setup;
+        let (block_conv_sender, block_conv_receiver) = mpsc::channel(100);
+        let block_importer =
+            Arc::new(BlockImporter::new(backend.clone(), &MetricsRegistry::dummy(), None, true).unwrap());
+        let validation = BlockValidationContext::new(backend.chain_config().chain_id.clone());
+        let telemetry = TelemetryService::new(true, vec![]).unwrap().new_handle();
+
+        let mock_block = UnverifiedFullBlock {
```
Comment (Mohiiit) [inline crates/client/sync/src/l2.rs]: same, can have a helper function

**Review by Mohiiit — COMMENTED** [review]
Very clean, just a few changes

**Review by Mohiiit — APPROVED** [review]

File: crates/client/block_import/Cargo.toml [hunk 3]
```diff
@@ -8,6 +8,9 @@ repository.workspace = true
 version.workspace = true
 license.workspace = true
 
+[features]
+default = []
+testing = []
```
Comment (antiyro) [inline crates/client/block_import/Cargo.toml:13]: what is this for?

**Review by EvolveArt — APPROVED** [review]

File: crates/client/block_import/Cargo.toml (hunk 3 above)
Comment (EvolveArt) [reply crates/client/block_import/Cargo.toml:13]: for some utils only under testing feature

**Pull request merged**